cpp/vm/evmzero/memory.h

```diff
@@ -70,6 +70,14 @@ class Memory {
     std::copy_n(memory_.data() + memory_offset, buffer.size(), buffer.data());
   }
 
+  // Reads from memory, writes into memory, grows memory if necessary.
+  void MemCopy(uint64_t src_offset, uint64_t dest_offset, uint64_t size) {
+    Grow(src_offset, size);
+    Grow(dest_offset, size);
+    // copy with overlap guarantees
+    std::memmove(memory_.data() + dest_offset, memory_.data() + src_offset, size);
```

Comment:
Can you add unit test for this function into [memory_test.cc](https://github.com/Fantom-foundation/Tosca/blob/main/cpp/vm/evmzero/memory_test.cc)? 
In particular it should check that overlapping memory regions are handled correctly.